### PR TITLE
Enforce magnifications via task type

### LINKED ISSUE #4075: Allow to enforce specific mag(s) for tasks
If the user is zoomed out too far, a toast should appear which says "please zoom in".

### PR DESCRIPTION
**Pull request by philippotto**
@youri-k Can you update the migration guide? A datastore update is probably not necessary, is it?

Additionally, to the mag enforcement, I also added an "active resolution" hint to the info tab. I think, this is quite useful (at least, it was for testing :)).

### URL of deployed dev instance (used for testing):
- https://enforcemags.webknossos.xyz

### Steps to test:
- edit a task type and check "enforce magnifications"
- fill min and max (you can also try to omit one value, as this should work, too)
- create a task for that task type
- trace the task and ensure that you can only zoom within that range

### Issues:
- fixes #4075 

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- [x] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- ~[ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~
- [ ] Needs datastore update after deployment
- [X] Ready for review

### CODE REVIEW

File: frontend/javascripts/admin/tasktype/task_type_create_view.js [hunk 1]
```diff
@@ -103,11 +159,11 @@ class TaskTypeCreateView extends React.PureComponent<Props, State> {
   render() {
     const { getFieldDecorator } = this.props.form;
     const isEditingMode = this.props.taskTypeId != null;
-    const titlePrefix = isEditingMode ? "Update " : "Create";
+    const titlePrefix = isEditingMode ? "Update" : "Create";
```
Comment (daniel-wer) [inline frontend/javascripts/admin/tasktype/task_type_create_view.js]: :eagle: :eyes:

File: frontend/javascripts/oxalis/model/accessors/dataset_accessor.js [hunk 2]
```diff
@@ -299,4 +299,15 @@ export function getSegmentationThumbnailURL(dataset: APIDataset): string {
   return "";
 }
 
+function _keyResolutionsByMax(dataset: APIDataset): { number: Vector3 } {
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/model/accessors/dataset_accessor.js]: The return type should probably be `{ [number]: Vector3 }`

File: frontend/javascripts/admin/tasktype/task_type_create_view.js [hunk 3]
```diff
@@ -33,6 +45,33 @@ type State = {
   useRecommendedConfiguration: boolean,
 };
 
+function isValidMagnification(rule, value, callback) {
+  if (value === "" || value == null || (Math.log(value) / Math.log(2)) % 1 === 0) {
+    callback();
+  } else {
+    callback("The magnification must be stated as a power of two (e.g., 1 or 2 or 4 or 8 ...");
```
Comment (daniel-wer) [inline frontend/javascripts/admin/tasktype/task_type_create_view.js]:
```suggestion
    callback("The magnification must be stated as a power of two (e.g., 1 or 2 or 4 or 8 ...)");
```

**Comment by philippotto** [comment]
> I started testing, but encountered an issue right away, so I first wanted to let you check that before continuing the testing:
> 
> * I created a task type with a min resolution of 4 and a max of 512
> * I then created a task with that task type and opened said task
> * The task loads at resolution 2-2-2 and I cannot zoom (neither in nor out)

Oh, very good catch! I fixed it now.

Also, do you notice any performance regressions when zooming? Locally, it appeared a bit laggier to me, but the reason might as well be that I used wk predominantly on deployed instances (where the code is optimized by compilation). In benchmarks, the code wasn't conspicuous.

**Comment by philippotto** [comment]
> Backend LGTM (please update the PR number in evolution file + add migration guide entry)

@youri-k Could you do this? :)

**Comment by daniel-wer** [comment]
Works very well!
I didn't notice any performance regression when zooming, I'm also getting similar FPS during zooming on master and this branch :)

**Pull request merged**